test/side_effects_test.cpp

```diff
@@ -76,3 +76,49 @@ TEST_CASE("side_effects_test - execute_parallelly_atomic")
     execute_parallelly(replicate(4, inc_atomic_int_return_true))();
     REQUIRE_EQ(atomic_int.load(), 4);
 }
+
+template<int rows, int cols>
+struct MandelbrotMat
+{
+    std::vector<int> nb_iterations;
+    MandelbrotMat() : nb_iterations(rows * cols, 0) {}
+};
+
+TEST_CASE("side_effects_test - parallel_for")
+{
+    using namespace fplus;
+
+    constexpr int rows = 3000, cols = 2000;
```

Comment:
If I understand correctly, on a 64-bit system, this will use 48 MB. How about a smaller, maybe even generally simpler, test?